src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -358,6 +358,10 @@ pub fn compile<'cx>(cx: &'cx mut ExtCtxt,
         valid &= check_rhs(cx, rhs);
     }
 
+    for lhs in &lhses {
+        valid &= check_lhs_no_empty_seq(cx, &[lhs.clone()])
+    }
```

Comment:
ok, good point on not reporting all errors. Could you add a comment to explain that please?